src/util/cxxlib.h

```diff
@@ -33,6 +33,222 @@
 
 namespace upx {
 
+/*************************************************************************
+// compile_time
```

Comment:
Why are there so many lines of diff?  Explain.  Such an un-motivated,  large change is impossible to review.  And  Is the intent to have "No functionsl change" (`NFC`), or not? 
Split the change into two separate changes: One with any functional changes (explained, and with no gross re-organization), then one which is only moving the lines around (and commented as `NFC`).